### PR TITLE
[pallet-revive] move event topics in event's body

### PR DESCRIPTION
**Pull request by pgherveou**
Fix https://github.com/paritytech/polkadot-sdk/issues/5629

### CODE REVIEW

File: substrate/frame/revive/src/lib.rs [hunk 1]
```diff
@@ -406,6 +406,9 @@ pub mod pallet {
 			/// Data supplied by the contract. Metadata generated during contract compilation
 			/// is needed to decode it.
 			data: Vec<u8>,
+			/// A list of topics used to index the event.
+			/// Number of topics is capped by [`limits::NUM_EVENT_TOPICS`].
```
Comment (pgherveou) [inline substrate/frame/revive/src/lib.rs:410]: Should we just hard code the max to 4, as this is what you can have in EVM 🤔

**Comment by athei** [comment]
R and T labels are mutually exclusive AFAIK.

File: substrate/frame/revive/src/lib.rs (hunk 1 above)
Comment (athei) [reply substrate/frame/revive/src/lib.rs:410]: Using a constant is already hard coding. How would you hard code it harder? Using a `BoundedVec`?

**Review by athei — APPROVED** [review]

**Review by xermicus — APPROVED** [review]

File: substrate/frame/revive/src/lib.rs (hunk 1 above)
Comment (pgherveou) [reply substrate/frame/revive/src/lib.rs:410]: ah nvm I though it was configurable for some reason

**Pull request merged**